File: frontend/stock_app/lib/domain/entities/categoria.dart
class Categoria {
  final String id;
  final String nombre;
  final String? descripcion;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Categoria({
    required this.id,
    required this.nombre,
    this.descripcion,
    this.createdAt,
    this.updatedAt,
  });

  Categoria copyWith({
    String? id,
    String? nombre,
    String? descripcion,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Categoria(
      id: id ?? this.id,
      nombre: nombre ?? this.nombre,
      descripcion: descripcion ?? this.descripcion,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}

File: backend/internal/model/categoria.go
package model

import (
	"time"
)

type Categoria struct {
	ID          string    `json:"id"`
	Nombre      string    `json:"nombre"`
	Descripcion *string   `json:"descripcion"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type CreateCategoriaRequest struct {
	Nombre      string  `json:"nombre" validate:"required"`
	Descripcion *string `json:"descripcion"`
}

type UpdateCategoriaRequest struct {
	Nombre      *string `json:"nombre"`
	Descripcion *string `json:"descripcion"`
}